### ISSUE TITLE
Add authentication for the test report page

### ISSUE THREAD

**Issue by anoop2811** [Guidewire/fern-reporter#34]
The client should be using an auth scheme for sending information to the fern server
The creds could be used to identify the authorized test projects

**Comment by jjchambl** [comment]
Are we thinking token-based/oidc integration? Do we need to tie this to the reports table in the db?

**Comment by anoop2811** [comment]
@jjchambl : Yes, we are thinking okta integration and we would need to tie it to the reports in the db such that only the ones that I am authorized to see would be visible to me. Check with Bharath Sekar for more details, he may have ideas.

**Comment by bsekar** [comment]
![image](https://github.com/Guidewire/fern-reporter/assets/20427292/82e56ff4-50a6-42b1-9f1c-cf502a91f32c)

**Comment by bsekar** [comment]
Ensure a follow-up for GRPC based report creation for performance reasons.

**Comment by marius-williams** [comment]
@anoop2811 and @bsekar, I would like to pick this up. Is this issue's information current, based on the current state of fern, or would this issue need to be reworked?
